apis/python/src/tiledbsoma/io/outgest.py

```diff
@@ -472,6 +522,10 @@ def _extract_uns(
     for key, value in collection.metadata.items():
         if level == 0 and uns_keys is not None and key not in uns_keys:
             continue
+        if key in extracted:
+            # This should not be possible, as it implies we wrote the same key to the collection and metadata.
+            # Just skip if it happens, as a precaution.
+            continue
```

Comment:
Should this error out or give some sort of warning?

Reply:
metaphysical question as it implies it is a bug - I'm not sure it is as we don't have clear semantics defined for uns  when a user modifies it to contain duplicate keys (e.g., if a user ingests an H5AD, the manually modifies the `uns` data using the SOMA API, is that a bug)?

Previously the code treated the contents of `metadata` as having precedence over the Collection contents.  This code simply reverses that precedence (which in retrospect, is probably a backward-compat breaking change).

@johnkerl - do you have an opinion which should have precedence in the case that both `uns.metadata["key"]` and `uns["key"]` exist?  I think the options are:
* generate error 
* metadata wins
* collection wins
Previous behavior was "metadata wins".

it can definitely happen with post-ingest modification by a user.

Reply:
BTW, I'm going to push a commit that reverts to the historical behavior (i.e., doesn't break backward compat).  But the open question remains - what we _want_ to happen.

Reply:
@bkmartinjr I think 'keep same' is best -- whatever we may or may not have documented, I think:

* switching the order will break no one and no one will notice; or
* for someone for whom the order matters in ways they might not even realize in their data pipelines, something weird may start happening

Reply:
Concur - just pushed the mod.